packages/xo-server/src/xo-mixins/remotes.mjs

```diff
@@ -297,4 +298,16 @@ export default class {
 
     await this._remotes.remove(id)
   }
+
+  async getTotalBackupSizeOnRemote(id) {
+    const remote = await this._getRemote(id)
+
+    if (remote.proxy !== undefined) {
+      return this._app.callProxyMethod(remote.proxy, 'remote.getTotalBackupSize', {
+        remote,
+      })
+    }
+
+    return Disposable.use(this._app.getBackupsRemoteAdapter(remote), adapter => adapter.getTotalBackupSize())
```

Comment:
disposable are unneeded here since XO already handles a cache of remoteadapter, and reusing the already instantiated one if possible